Update AssemblyBuilder.md with persistable AssemblyBuilder info (#39491)
dotnet/docs: docs/core/porting/net-framework-tech-unavailable.md

> [!TIP]
> Windows Communication Foundation (WCF) server can be used in .NET 6+ by using the [CoreWCF NuGet packages](https://www.nuget.org/profiles/corewcf). For more information, see [CoreWCF 1.0 has been Released](https://devblogs.microsoft.com/dotnet/corewcf-v1-released/).

## Some reflection emit APIs are not supported

> .NET 8 and below doesn't support saving assemblies that are generated by the <xref:System.Reflection.Emit?displayProperty=fullName> APIs. The <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWithType> method is not available in .NET 8 and below. In addition, the following fields of the <xref:System.Reflection.Emit.AssemblyBuilderAccess> enumeration aren't available:

Reviewer: ```suggestion
.NET 8 and earlier versions of .NET (Core) don't support saving assemblies that are generated by the <xref:System.Reflection.Emit?displayProperty=fullName> APIs, and the <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWithType> method isn't available. In addition, the following fields of the <xref:System.Reflection.Emit.AssemblyBuilderAccess> enumeration aren't available:
```

- <xref:System.Reflection.Emit.AssemblyBuilderAccess.ReflectionOnly>
- <xref:System.Reflection.Emit.AssemblyBuilderAccess.RunAndSave>
- <xref:System.Reflection.Emit.AssemblyBuilderAccess.Save>

> In .NET 9.0 we added new persisted `AsemblyBuilder` implementation and added the <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWithType> method back. Check [this doc](../../fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md) to learn more about how to use the new API.

Reviewer: ```suggestion
In .NET 9, a persisted `AssemblyBuilder` was implemented and the <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWithType> method was added back to the reflection emit library. To learn more about how to use this API, see [System.Reflection.Emit.AssemblyBuilder class](../../fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md).
```

> In .NET 9, a persisted `AssemblyBuilder` was implemented and the <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWithType> method was added back to the reflection emit library. To learn more about how to use this API, see [System.Reflection.Emit.AssemblyBuilder class](../../fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md).

Author: ```suggestion
In .NET 9, a persisted `AssemblyBuilder` was implemented and the <xref:System.Reflection.Emit.AssemblyBuilder.Save%2A?displayProperty=nameWithType> method was added back to the reflection emit library. To learn more about how to use this API, see [System.Reflection.Emit.AssemblyBuilder class](../../fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md#persistable-dynamic-assemblies-in-net-core).
```
